### PR TITLE
Fix doc generation for the root configuration

### PR DESCRIPTION
**Pull request by pvalsecc**
Partial fix for #666 (the number of the beast)

### CODE REVIEW

**Comment by sbrunner** [comment]
Does it also check:
- that the given arguments are correct?
- that we don't have extra attribute at the root level?

**Comment by pvalsecc** [comment]
I've made the descrition of `throwErrorOnExtraParameters` more accurate.

File: core/src/main/java/org/mapfish/print/config/Configuration.java [hunk 1]
```diff
@@ -449,17 +454,28 @@ public final void setDefaultStyle(final Map<String, Style> defaultStyle) {
...
+     * @param throwErrorOnExtraParameters the value
+     */
     public final void setThrowErrorOnExtraParameters(final boolean throwErrorOnExtraParameters) {
         this.throwErrorOnExtraParameters = throwErrorOnExtraParameters;
     }
 
+    @Override
+    public void validate(final List<Throwable> validationErrors, final Configuration configuration) {
+        validationErrors.addAll(validate());
+    }
```
Comment (sbrunner) [inline core/src/main/java/org/mapfish/print/config/Configuration.java]: What's this for?

Comment (pvalsecc) [reply core/src/main/java/org/mapfish/print/config/Configuration.java]: I need to implement the `ConfigurationObject` for the class to be included in the doc and this method is requested by this interface.

Comment (sbrunner) [reply core/src/main/java/org/mapfish/print/config/Configuration.java]: OK, thanks :-)

**Pull request merged**